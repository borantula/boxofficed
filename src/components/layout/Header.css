.site-title {
  margin: 0;
  font-family: "Cinzel", "Times New Roman", Arial, serif;
  font-weight: 400;
  letter-spacing: 4px;
  color: #000000;
  text-decoration: none;
}
.site-title:hover {
  color: #333333;
}

.my-list-btn {
  color: #000000;
  text-decoration: none;
}
