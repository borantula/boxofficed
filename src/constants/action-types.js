//TODO name things in more logical groups like API requests, filtering etc
export const UPDATE_MOVIE_LIST = "UPDATE_MOVIE_LIST";
export const REQUEST_MOVIES_BY_YEAR = "REQUEST_MOVIES_BY_YEAR";
export const CHANGE_YEAR = "CHANGE_YEAR";
export const CHANGE_GENRE = "CHANGE_GENRE";
export const DISPLAY_MOVIE = "DISPLAY_MOVIE";
export const RESET_DISPLAYED_MOVIE = "RESET_DISPLAYED_MOVIE";
export const GET_GENRE_LIST = "GET_GENRE_LIST";
export const USER_LOGGED_IN = "USER_LOGGED_IN";
export const USER_LOGGED_OUT = "USER_LOGGED_OUT";
export const ADD_MOVIE_TO_SAVED_LIST = "ADD_MOVIE_TO_SAVED_LIST";
export const REMOVE_MOVIE_FROM_SAVED_LIST = "REMOVE_MOVIE_FROM_SAVED_LIST";
export const UPDATE_SAVED_MOVIES_LIST_FROM_SERVER =
  "UPDATE_SAVED_MOVIES_LIST_FROM_SERVER";
