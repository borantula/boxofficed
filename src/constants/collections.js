export const COLLECTION_SAVEDLIST = "savedlist";
