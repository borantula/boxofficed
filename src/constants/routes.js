export const HOME = "/";
export const MOVIE = "/movie/:movieId/:movieSlug";
export const SIGNIN = "/signin";
export const MYLIST = "/mylist";
