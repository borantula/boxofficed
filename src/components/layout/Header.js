import React from "react";
import PropTypes from "prop-types";
import "./Header.css";

const Header = ({ title = "", checkUser }) => {
  return (
    <header className="header">
      <h1>
        <a href="/">{title}</a>
      </h1>
    </header>
  );
};

Header.propTypes = {
  title: PropTypes.string,
};

export default Header;
