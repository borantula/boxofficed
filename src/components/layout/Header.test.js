import React from "react";
import ReactDOM from "react-dom";
import Header from "./Header";
import * as routes from "../../constants/routes";
import { shallow } from "enzyme";

const mockUser = {
  isLoggedIn: false,
};

describe("Header without user and empty bag", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(
      <Header routes={routes} savedMoviesCount={0} user={mockUser} debug />
    );

    expect(component).toMatchSnapshot();
  });
});

describe("Header without user and full bag", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(
      <Header routes={routes} savedMoviesCount={10} user={mockUser} debug />
    );
    expect(component).toMatchSnapshot();
  });
});
