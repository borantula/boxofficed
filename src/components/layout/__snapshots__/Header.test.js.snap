// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header without user and empty bag should render correctly in "debug" mode 1`] = `
<Header
  classes={
    Object {
      "appbar": "Header-appbar-3",
      "grow": "Header-grow-2",
      "root": "Header-root-1",
    }
  }
  debug={true}
  routes={
    Object {
      "HOME": "/",
      "MOVIE": "/movie/:movieId/:movieSlug",
      "MYLIST": "/mylist",
      "SIGNIN": "/signin",
    }
  }
  savedMoviesCount={0}
  user={
    Object {
      "isLoggedIn": false,
    }
  }
/>
`;

exports[`Header without user and full bag should render correctly in "debug" mode 1`] = `
<Header
  classes={
    Object {
      "appbar": "Header-appbar-3",
      "grow": "Header-grow-2",
      "root": "Header-root-1",
    }
  }
  debug={true}
  routes={
    Object {
      "HOME": "/",
      "MOVIE": "/movie/:movieId/:movieSlug",
      "MYLIST": "/mylist",
      "SIGNIN": "/signin",
    }
  }
  savedMoviesCount={10}
  user={
    Object {
      "isLoggedIn": false,
    }
  }
/>
`;
